### PR TITLE
Refactor Scale

### LINKED ISSUE #1628: Refactor initScale
When possible try to do as many things as possible in `initScale` instead of `parseScale` so that the rest of the parse process depends less on the `parseScale`

### PR DESCRIPTION
**Pull request by kanitw**
Fix #1628

## TO DO

- [x] Check why some stuff are now broken in the gallery
  - [x] Faceting
    - [x] Test Faceting
  - [x] Consider if we will allow padding for row / column

- [x] Update docs

## TO FOLLOW UP

-  See if we can make band scale always work for bar (#1616)
- Should `parseMainScale` separate `scaleSignature` from scaleDomain? (#1645)

### CODE REVIEW

File: src/compile/scale.ts [hunk 1]
```diff
@@ -44,11 +44,7 @@ export type ScaleComponents = {
 }
 
 export function initScale(topLevelSize: number, mark: Mark, channel: Channel, fieldDef: ChannelDefWithScale, scaleConfig: ScaleConfig): Scale {
-  let scale: Scale = extend({
-    round: scaleConfig.round,
-    useRawDomain: scaleConfig.useRawDomain, // TODO determine if we should keep this
-    padding: scaleConfig.padding // TODO determine if we should keep this
-  }, (fieldDef || {}).scale);
+  let scale: Scale = extend({}, (fieldDef || {}).scale);
```
Comment (domoritz) [inline src/compile/scale.ts]: is this just to make a copy?

Comment (kanitw) [inline src/compile/scale.ts]: yeah feel free to change to duplicate

File: src/scale.ts [hunk 2]
```diff
@@ -57,14 +57,26 @@ export interface ScaleConfig {
...
+   * @minimum 0
+   * @maximum 1
    */
   pointPadding?: number;
 
   /**
+   * Default spacing between faceted plots.
+   * @type {integer}
+   * @minimum 0
+   * @maximum 1
```
Comment (domoritz) [inline src/scale.ts]: Should not have max

Comment (domoritz) [inline src/scale.ts]: Sound not have max

File: src/compile/scale.ts [hunk 3]
```diff
@@ -49,13 +49,13 @@ export function initScale(topLevelSize: number, mark: Mark, channel: Channel, fi
   scale.type = initType(scale.type, fieldDef, channel, mark);
   if (scale.type === ScaleType.ORDINAL) {
     // TODO: if possible, make points (ordinalType) not dependent on bandSize
-    const bandSize = initBandSize(scale.bandSize, topLevelSize, mark, channel, scaleConfig);
-    if (bandSize !== undefined) {
-      scale.bandSize = bandSize;
+    const _bandSize = bandSize(scale.bandSize, topLevelSize, mark, channel, scaleConfig);
+    if (_bandSize !== undefined) {
```
Comment (domoritz) [inline src/compile/scale.ts]: _bandSize -> size

File: test/compile/scale.test.ts [hunk 4]
```diff
@@ -88,37 +88,37 @@ describe('Scale', function() {
     });
   });
 
-  describe('initBandSize()', () => {
+  describe('bandSize()', () => {
 
     it('should return undefined if bandSize spec is fit', () => {
-      const bandSize = initBandSize(BANDSIZE_FIT, 180, POINT, X, defaultScaleConfig);
-      assert.deepEqual(bandSize, undefined);
+      const _bandSize = bandSize(BANDSIZE_FIT, 180, POINT, X, defaultScaleConfig);
```
Comment (domoritz) [inline test/compile/scale.test.ts]: _bandSize -> size

File: vega-lite-schema.json [hunk 5]
```diff
@@ -1668,10 +1672,24 @@
...
+                    "description": "Default padding for `x` and `y` point-ordinal scales.",
+                    "minimum": 0,
+                    "maximum": 1,
+                    "type": "number"
+                },
+                "facetSpacing": {
+                    "description": "Default spacing between faceted plots.",
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 1
```
Comment (domoritz) [inline vega-lite-schema.json]: need to rebuild

**Pull request merged**